### PR TITLE
Draggable numbers - fix undo, particle angles, being disabled, more robust dragging

### PR DESCRIPTION
**Pull request by dback2**
Addresses:
https://highfidelity.manuscript.com/f/cases/19978/
https://highfidelity.manuscript.com/f/cases/19997/
https://highfidelity.manuscript.com/f/cases/19998/
https://highfidelity.manuscript.com/f/cases/20085/

Also fixes draggable numbers not being disabled when properties are all disabled, and incorrect label on Spin property.

Pre-Merge Test Plan:
1. Enter Interface in desktop mode into any non-sandbox domain with edit permissions
2. Open Create and create or select any Particle entity
3. Verify if quickly clicking and dragging on vector properties such as Position or Rotation and releasing when over a neighboring value (such as in bug 19978 above) that the neighboring value does not get focused (text highlighted and arrows disappear). This should probably be tested many times as it was hard to repro.
4. Verify the Horizontal Angle Start and Finish properties have ranges of 0 to 180 and the Vertical Angle Start and Finish properties have a range of -180 to 180. Verify that editing all 4 values by dragging them works as expected.
5. Verify if quickly clicking and dragging on numerical particle properties (such as Spin or Size) to perform a very fast change that upon releasing the displayed value stays the same. Also verify if dragging numerical particle properties back and forth very quickly for a few seconds that upon releasing the displayed value stays the same. 
**Mac Note:** It is expected behavior on Mac that some numerical particle properties might lag in this case where the value changes back and forth quickly after releasing (especially if framerate is low), and this behavior is the same as in RC75 on Mac with the old sliders for these values.
6. Perform step 5 again but instead on numerical non-particle properties (such as Dimensions or Local Position).
**Mac Note:** Unlike in step 5, the non-particle properties should not have the delayed update after releasing on Mac.
7. Perform many subsequent drags of multiple different numerical properties, and verify that each undo action will undo each individual value change from drag start to drag finish, and verify that each redo performs as expected.
**Mac Note:** It is expected behavior on Mac that multiple undo actions can be triggered if Cmd + Z keys are held down. This PR, however, should fix the fact that an undo was being triggered on both pressing Cmd + Z and releasing Cmd + Z.
8. Create any new entity and repeat step 7 on the new entity including undoing changes all the way up to deleting the entity.
9. Select a locked entity or lock an entity and verify that attempting to use the arrows on any draggable numbers or clicking to drag them does nothing.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15331/HighFidelity-Beta-PR14456-5095352.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15336/HighFidelity-Beta-PR14456-5095352.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15336/HighFidelity-Beta-PR14456-5095352.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15336/HighFidelity-Beta-Interface-PR14456-5095352.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15336/HighFidelity-Beta-PR14456-5095352.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15336/HighFidelity-Beta-Interface-PR14456-5095352.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15339/HighFidelity-Beta-PR14456-c077734.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15339/HighFidelity-Beta-PR14456-c077734.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15339/HighFidelity-Beta-Interface-PR14456-c077734.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15339/HighFidelity-Beta-PR14456-c077734.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15339/HighFidelity-Beta-Interface-PR14456-c077734.exe

**Comment by jakram** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15374/HighFidelity-Beta-PR14456-c077734.apk

File: scripts/system/edit.js [hunk 1]
```diff
@@ -2360,10 +2361,16 @@ var PropertiesTool = function (opts) {
...
-            blockPropertyUpdates = data.blockUpdateCallback === true;
+            if (data.onlyUpdateEntities) {
+                blockPropertyUpdates = data.blockUpdateCallback;
+            } else {
+                pushCommandForSelections();
+                SelectionManager.saveProperties();
+            }
             selectionManager._update(false, this);
-            blockPropertyUpdates = false;
+            if (data.onlyUpdateEntities) {
```
Comment (huffman) [inline scripts/system/edit.js]: Is this if necessary? Don't we always want to go back to `false`?

File: scripts/system/html/js/draggableNumber.js [hunk 2]
```diff
@@ -44,32 +44,33 @@ DraggableNumber.prototype = {
     },
     
     documentMouseMove: function(event) {
-        if (this.initialMouseEvent) {
+        if (!this.dragging && this.initialMouseEvent) {
```
Comment (huffman) [inline scripts/system/html/js/draggableNumber.js]: I think we should check the `dragging` value inside of the mouseUp function. If you end up dragging back near where you had started, you can trigger the logic to focus on the input field.

**Comment by JeffClinton** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15405/HighFidelity-Beta-PR14456-c077734.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15405/HighFidelity-Beta-PR14456-c077734.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15405/HighFidelity-Beta-Interface-PR14456-c077734.dmg

**Comment by qligmaloney** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15419/HighFidelity-Beta-PR14456-c077734.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15419/HighFidelity-Beta-PR14456-c077734.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15419/HighFidelity-Beta-Interface-PR14456-c077734.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15419/HighFidelity-Beta-PR14456-c077734.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15419/HighFidelity-Beta-Interface-PR14456-c077734.exe

**Comment by QliCknox** [comment]
Currently Testing

**Comment by QliCknox** [comment]
Fail @conklin94122 

Step 4 - Test plan states "Verify the Horizontal and Vertical Angle Start and Finish properties have ranges of 0 to 180, except for Vertical Angle Start which was a range of -180 to 0."

Vertical Angle is only able to be slid from -180 to 0. But you can manually put in a positive number like 100 and press enter and the properties tab will keep this value. If you set anything above 180, it will rest to 180. 

Secondly, Vertical Angle Finish is able to do the opposite - it starts at 180 and should go to 0. But you can manually put negative numbers down to -180 and it will keep these values. 

Clarifying question for step 5 - Quickly dragging a property will make the text act weird and jump around. If it looks like I landed on 2.14, it may jump to 2 instead. Is that what you're expecting? Changing it quickly makes it hard to pin down what the properties number should be.

When undoing these actions, it also looks like the numbers are taking time to update and the user sees other values flash as they undo to the previously changed value

Also, please note in the first second of the video I selected the create tools window the the property changes from 2.55 to 2.56 without touching the property value

The above three issues are shown in the video below. Log included

[Archive.zip](https://github.com/highfidelity/hifi/files/2621363/Archive.zip)

File: scripts/system/edit.js (hunk 1 above)
Comment (dback2) [reply scripts/system/edit.js]: True we do yes, it's not necessary

File: scripts/system/html/js/draggableNumber.js (hunk 2 above)
Comment (dback2) [reply scripts/system/html/js/draggableNumber.js]: Good point did not notice that

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15466/HighFidelity-Beta-PR14456-887929e.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15466/HighFidelity-Beta-PR14456-887929e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15466/HighFidelity-Beta-Interface-PR14456-887929e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15466/HighFidelity-Beta-PR14456-887929e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15466/HighFidelity-Beta-Interface-PR14456-887929e.exe

**Comment by dback2** [comment]
Hopefully QA fail cases #2 (value snapping on release) and #3 (undo) are now addressed

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15471/HighFidelity-Beta-PR14456-c19067c.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15471/HighFidelity-Beta-PR14456-c19067c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15471/HighFidelity-Beta-Interface-PR14456-c19067c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15471/HighFidelity-Beta-PR14456-c19067c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15471/HighFidelity-Beta-Interface-PR14456-c19067c.exe

**Comment by QliCknox** [comment]
Currently Testing

**Comment by QliCknox** [comment]
Fail @conklin94122 

Undoing an action now undos more than one change for a singular property - as tested on a Particles Physics. Undo also undos other properties the user changes, as tested on a cubes physical section
[Archive.zip](https://github.com/highfidelity/hifi/files/2625907/Archive.zip)

**Comment by dback2** [comment]
QA failure 1 from above with undo causing an undo of more than value change is Mac specific and being investigated

QA failure 2 from above may be specific behavior to the Local Rotation property (where all 3 vector values can change from an individual value change) and will be better tested on other physical properties

**Comment by dback2** [comment]
The issue where pressing Cmd + Z and releasing Cmd + Z both triggering undo on Mac should now be fixed. Notes have been added to the test plan for the expected behavior on Mac in regards to multiple undo actions due to Cmd + Z being held, and for particle property updates to sometimes lag.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15493/HighFidelity-Beta-PR14456-7e7ed8f.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15494/HighFidelity-Beta-PR14456-9c2cbba.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15494/HighFidelity-Beta-PR14456-9c2cbba.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15494/HighFidelity-Beta-Interface-PR14456-9c2cbba.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15494/HighFidelity-Beta-PR14456-9c2cbba.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15494/HighFidelity-Beta-Interface-PR14456-9c2cbba.exe

**Comment by jakram** [comment]
Currently testing

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15513/HighFidelity-Beta-PR14456-9c2cbba.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15513/HighFidelity-Beta-PR14456-9c2cbba.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15513/HighFidelity-Beta-Interface-PR14456-9c2cbba.dmg

**Pull request merged**